kroxylicious-test-tools/src/main/java/io/kroxylicious/test/tester/KroxyliciousClients.java

```diff
@@ -64,14 +77,27 @@ public Consumer<String, String> consumer() {
 
     public <U, V> Consumer<U, V> consumer(Serde<U> keySerde, Serde<V> valueSerde, Map<String, Object> additionalConfig) {
         Map<String, Object> config = createConfigMap(bootstrapServers, additionalConfig);
-        return CloseableConsumer.wrap(new KafkaConsumer<>(config, keySerde.deserializer(), valueSerde.deserializer()));
+        Consumer<U, V> consumer = CloseableConsumer.wrap(new KafkaConsumer<>(config, keySerde.deserializer(), valueSerde.deserializer()));
+        consumers.add(consumer);
+        return consumer;
     }
 
     public KafkaClient simpleTestClient() {
         String[] hostPort = bootstrapServers.split(":");
         return new KafkaClient(hostPort[0], Integer.parseInt(hostPort[1]));
     }
 
+    public void close() {
+        try {
+            admins.forEach(Admin::close);
```

Comment:
Is close idempotent? (I would hope so but can't remember) if a user has closed the client already is the tester calling close again going to cause issues?

Reply:
There's no straightforward way to check if a client has already been closed that I could see, so it's making the assumption that they haven't been. I'll see about adding a test case to cover that scenario, if it's problematic then I'll have to revisit if there's a way to check for this.